### PR TITLE
Use settings.get_setting instead of direct attribute access

### LINKED ISSUE #612: Clean up module use of settings user API
Accessing `settings.foo.bar` is a user API, not intended for modules. In places like `lib/window.lua`, `settings.get_setting()` should be used instead.

### PR DESCRIPTION
**Pull request by McSinyx**
To fix #612.

### CODE REVIEW

File: lib/window.lua [hunk 1]
```diff
@@ -863,7 +863,7 @@ settings.migrate_global("window.max_title_len", "max_title_len")
 
 local globals = package.loaded.globals
 if globals then
-    settings.window.search_engines = globals.search_engines
+    settingsi.window.search_engines = globals.search_engines
```
Comment (aidanholm) [inline lib/window.lua]: whoops :)

Comment (aidanholm) [inline lib/window.lua]: whoops :)

**Review by aidanholm — CHANGES_REQUESTED** [review]
Minor typo

**Comment by McSinyx** [comment]
I'm so sorry. I got that fixed now...

**Comment by aidanholm** [comment]
No worries, happens to all of us :) this looks great, thanks a lot!

**Pull request merged**